### ISSUE TITLE
Unable to see docker related information

### ISSUE THREAD

**Issue by viticis** [nicolargo/glances#2180]
**Describe the bug**
Unable to see docker related information

**To Reproduce**

1. 
```yaml
services:
  glances:
    image: nicolargo/glances
    container_name: glances
    restart: always
    environment:
      - GLANCES_OPT=-w
    pid: host
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
```
2. 
```bash
docker compose up -d
```

**Expected behavior**
Enable to see docker related information

**Screenshots**
![image](https://user-images.githubusercontent.com/28482172/199479672-19610c8a-225e-4b08-9322-a432c01b00a0.png)

**Desktop (please complete the following information):**
 - OS : Linux 5.18.0-0.bpo.1-cloud-amd64 #1 SMP PREEMPT_DYNAMIC Debian 5.18.2-1~bpo11+1 (2022-06-14) x86_64 GNU/Linux
 - Glances Version : Glances v3.3.0.2 with PsUtil v5.9.3
 - Docker version : Docker version 20.10.21, build baeda1f
 - Glances logs file : 
 ```
2022-11-02 11:35:46,475 -- DEBUG -- Plugin docker update return {} in 1.3e-05 seconds
2022-11-02 11:35:46,476 -- DEBUG -- docker plugin - Docker export error 'containers'
```
- Glances test :
``` ============================================================================================================================================================================================================
Glances 3.3.0.2 (/usr/local/lib/python3.10/site-packages/glances/__init__.py)
Python 3.10.8 (/usr/local/bin/python)
PsUtil 5.9.3 (/usr/local/lib/python3.10/site-packages/psutil/__init__.py)
============================================================================================================================================================================================================
alert         [OK]    0.00002s [[1667389079.0, -1, 'WARNING', 'MEM', 83.80329212156268, 83.80329212156268, 83.80329212156268, 83.80329212156268, 1, [], '', 'memory_percent']]
amps          [OK]    0.00011s []
cloud         [OK]    0.00006s {}
connections   [OK]    0.00788s {'LISTEN': 2, 'ESTABLISHED': 0, 'SYN_SENT': 1, 'SYN_RECV': 0, 'initiated': 1, 'terminated': 1, 'nf_conntrack_count': 73.0, 'nf_conntrack_max': 8192.0, 'nf_conntrac
core          [OK]    0.00049s {'phys': 1, 'log': 1}
cpu           [OK]    0.00059s {'total': 94.7, 'user': 7.4, 'nice': 0.0, 'system': 3.9, 'idle': 87.7, 'iowait': 0.0, 'irq': 0.0, 'softirq': 0.5, 'steal': 0.5, 'guest': 0.0, 'guest_nice': 0.0, 't
diskio        [OK]    0.00047s key=disk_name [{'time_since_update': 2.0781145095825195, 'disk_name': 'vda', 'read_count': 0, 'write_count': 2, 'read_bytes': 0, 'write_bytes': 40960, 'key': 'disk
docker        [OK]    0.00008s []
folders       [OK]    0.00007s []
fs            [OK]    0.00074s key=mnt_point [{'device_name': '/dev/vda1', 'fs_type': 'ext4', 'mnt_point': '/etc/resolv.conf', 'size': 9667530752, 'used': 3174621184, 'free': 6061383680, 'percen
gpu           [OK]    0.00007s []
help          [OK]    0.00001s None
ip            [OK]    0.00007s {}
irq           [OK]    0.00062s key=irq_line [{'irq_line': 'LOC', 'irq_rate': 1323, 'key': 'irq_line', 'time_since_update': 2.038641929626465}, {'irq_line': '32_virtio0-output.0', 'irq_rate': 124
load          [OK]    0.00017s {'min1': 0.08, 'min5': 0.29, 'min15': 0.3, 'cpucore': 1}
mem           [OK]    0.00031s {'total': 1019731968, 'available': 164073472, 'percent': 83.9, 'used': 855658496, 'free': 164073472, 'active': 137052160, 'inactive': 639188992, 'buffers': 3698688
memswap       [OK]    0.00042s {'total': 0, 'used': 0, 'free': 0, 'percent': 0.0, 'sin': 0, 'sout': 0, 'time_since_update': 2.0848238468170166}
network       [OK]    0.00080s key=interface_name [{'interface_name': 'lo', 'alias': None, 'time_since_update': 2.024693727493286, 'cumulative_rx': 33100, 'rx': 200, 'cumulative_tx': 33100, 'tx'
now           [OK]    0.00002s 2022-11-02 11:38:02 UTC
percpu        [OK]    0.00015s key=cpu_number [{'key': 'cpu_number', 'cpu_number': 0, 'total': 100.0, 'user': 27.0, 'system': 9.0, 'idle': 0.0, 'nice': 0.0, 'iowait': 2.0, 'irq': 0.0, 'softirq':
ports         [OK]    0.00080s []
processcount  [OK]    0.04229s {'total': 103, 'running': 1, 'sleeping': 67, 'thread': 331, 'pid_max': 0}
processlist   [OK]    0.00020s key=pid [{'gids': pgids(real=0, effective=0, saved=0), 'memory_percent': 5.557965600623398, 'name': 'dockerd', 'nice': 0, 'cpu_times': pcputimes(user=2362.39, syst
psutilversion [OK]    0.00006s (5, 9, 3)
quicklook     [OK]    0.00044s {'cpu': 94.7, 'percpu': [{'key': 'cpu_number', 'cpu_number': 0, 'total': 100.0, 'user': 27.0, 'system': 9.0, 'idle': 0.0, 'nice': 0.0, 'iowait': 2.0, 'irq': 0.0, '
raid          [OK]    0.00004s {}
sensors       [OK]    0.00001s []
smart         [OK]    0.00011s {}
system        [OK]    0.00001s {'os_name': 'Linux', 'hostname': '890aecc0f82f', 'platform': '64bit', 'linux_distro': 'Debian GNU/Linux 10', 'os_version': '5.18.0-0.bpo.1-cloud-amd64', 'hr_name':
uptime        [OK]    0.00016s {'seconds': 253478}
wifi          [OK]    0.00007s []
============================================================================================================================================================================================================
Total time to update all stats: 0.07510s
============================================================================================================================================================================================================
```

**Additional context**
Thank you very much.

**Comment by viticis** [comment]
Yes! It works!

Although I don't know what's going on here.

I have seen similar issue happen constantly. 

Thank you very much for your time. 🙏🏻

**Comment by nicolargo** [comment]
In fact, the minimal image should include the Bottle (for WebUI) and Docker (for container monitoring) dependencies. But the it do not work... 

So the issue is not corrected. 

I will have a look on it to generate a minimal image with those two deps.

Thanks !

**Comment by dj-louw** [comment]
hello @nicolargo I came across this issue after trying to debug why I can't see docker container information while running Glances inside a docker container.

I updated my compose file to `image: nicolargo/glances:alpine-3.3.0.2-full` as suggested but I still am not seeing any docker information, either in the web-cli or exported to Influx.

I have tried:
* Running glances in `-w` mode, when opening up the web URL nothing happens when I press `^D`
* Running glances in `--export influxdb` mode, but I am not seeing any `docker` measurements coming through.

Is the issue that I am experiencing related to this thread, or is it something else? Then I will create a separate issue for it.

**Comment by dj-louw** [comment]
Hi, 

Please ignore my previous comment. Docker stats are now showing up after a system reboot (or something else, lots of things happened between me posting this message and seeing the working stats today).

**Comment by AlbertoLopSie** [comment]
I can't confirm that docker information is not showing on minimal image. I'm rebuilding it locally but cannot make it show

**Comment by AlbertoLopSie** [comment]
GOT IT !!!

minimal target on Dockerfile requires `py3-requests` to work:

``` yaml
##############################################################################
# minimal image
##############################################################################

# Create running images without any building dependency
FROM alpine:${IMAGE_VERSION} as minimal
ARG PYTHON_VERSION

RUN apk add --no-cache \
  python3 \
  py3-packaging \
  py3-dateutil \
  py3-requests \  <--- Add this to the `apk add'  command
  curl \
  lm-sensors \
  wireless-tools \
  iputils

COPY --from=buildRequirements /root/.local/bin /usr/local/bin/
COPY --from=buildRequirements /root/.local/lib/python${PYTHON_VERSION}/site-packages /usr/lib/python${PYTHON_VERSION}/site-packages/
COPY ./docker-compose/glances.conf /etc/glances.conf

# EXPOSE PORT (XMLRPC / WebUI)
EXPOSE 61209 61208

# Define default command.
WORKDIR /glances
CMD python3 -m glances -C /etc/glances.conf $GLANCES_OPT
```

Now is building and Docker info is working as a charm!!

![Screen Shot 2023-01-13 at 18 08 21](https://user-images.githubusercontent.com/2622480/212419505-9783a7ce-c291-44c4-a84c-9194386ec203.png)

**Comment by nicolargo** [comment]
Thanks @AlbertoLopSie !

Just pushed on the develop branch.

Will be available on Docker Hub images when Glances 3.3.1 will be released. Hope this week-end.
